From: Henry Su <henry.su@amd.com>
Date: Thu, 12 Jul 2007 12:12:29 +0000 (+0200)
Subject: i2c-piix4: Add support for the ATI SB700
X-Git-Tag: v2.6.23-rc1~1112^2~18
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=c29c22218b99dad95f7cd0281415a854aeee805c

i2c-piix4: Add support for the ATI SB700

Add the SMBus device ID for ATI SB700.

Signed-off-by: Henry Su <Henry.su@amd.com>
Signed-off-by: Jean Delvare <khali@linux-fr.org>
---

Backported to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.18.orig/Documentation/i2c/busses/i2c-piix4 linux-source-2.6.18/Documentation/i2c/busses/i2c-piix4
--- linux-source-2.6.18.orig/Documentation/i2c/busses/i2c-piix4	2007-08-03 02:39:13.196906000 -0600
+++ linux-source-2.6.18/Documentation/i2c/busses/i2c-piix4	2007-08-03 02:40:02.197618766 -0600
@@ -6,7 +6,7 @@ Supported adapters:
     Datasheet: Publicly available at the Intel website
   * ServerWorks OSB4, CSB5, CSB6 and HT-1000 southbridges
     Datasheet: Only available via NDA from ServerWorks
-  * ATI IXP200, IXP300, IXP400 and SB600 southbridges
+  * ATI IXP200, IXP300, IXP400, SB600 and SB700 southbridges
     Datasheet: Not publicly available
   * Standard Microsystems (SMSC) SLC90E66 (Victory66) southbridge
     Datasheet: Publicly available at the SMSC website http://www.smsc.com
diff -urpN linux-source-2.6.18.orig/drivers/i2c/busses/Kconfig linux-source-2.6.18/drivers/i2c/busses/Kconfig
--- linux-source-2.6.18.orig/drivers/i2c/busses/Kconfig	2007-08-03 02:39:13.196906000 -0600
+++ linux-source-2.6.18/drivers/i2c/busses/Kconfig	2007-08-03 02:40:02.197618766 -0600
@@ -177,6 +177,7 @@ config I2C_PIIX4
 	    ATI IXP300
 	    ATI IXP400
 	    ATI SB600
+	    ATI SB700
 	    Serverworks OSB4
 	    Serverworks CSB5
 	    Serverworks CSB6
diff -urpN linux-source-2.6.18.orig/drivers/i2c/busses/i2c-piix4.c linux-source-2.6.18/drivers/i2c/busses/i2c-piix4.c
--- linux-source-2.6.18.orig/drivers/i2c/busses/i2c-piix4.c	2007-08-03 02:39:13.196906000 -0600
+++ linux-source-2.6.18/drivers/i2c/busses/i2c-piix4.c	2007-08-03 02:40:02.197618766 -0600
@@ -23,7 +23,7 @@
    Supports:
 	Intel PIIX4, 440MX
 	Serverworks OSB4, CSB5, CSB6, HT-1000
-	ATI IXP200, IXP300, IXP400, SB600
+	ATI IXP200, IXP300, IXP400, SB600, SB700
 	SMSC Victory66
 
    Note: we assume there can only be one device, with one SMBus interface.
@@ -399,6 +399,8 @@ static struct pci_device_id piix4_ids[] 
 	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP600_SMBUS),
 	  .driver_data = 0 },
+	{ PCI_DEVICE(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP700_SMBUS),
+	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_SERVERWORKS, PCI_DEVICE_ID_SERVERWORKS_OSB4),
 	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_SERVERWORKS, PCI_DEVICE_ID_SERVERWORKS_CSB5),
diff -urpN linux-source-2.6.18.orig/include/linux/pci_ids.h linux-source-2.6.18/include/linux/pci_ids.h
--- linux-source-2.6.18.orig/include/linux/pci_ids.h	2007-08-03 02:39:13.196906000 -0600
+++ linux-source-2.6.18/include/linux/pci_ids.h	2007-08-03 02:40:02.197618766 -0600
@@ -365,6 +365,7 @@
 #define PCI_DEVICE_ID_ATI_IXP600_SMBUS	0x4385
 #define PCI_DEVICE_ID_ATI_IXP600_IDE	0x438c
 #define PCI_DEVICE_ID_ATI_IXP700_SATA	0x4390
+#define PCI_DEVICE_ID_ATI_IXP700_SMBUS	0x4395
 #define PCI_DEVICE_ID_ATI_IXP700_IDE	0x439c
 
 #define PCI_VENDOR_ID_VLSI		0x1004
